### DISCUSSION TITLE
Subsequent sign ins with the same provider fail with OAuthAccountNotLinked

### DISCUSSION THREAD

**Discussion by felixp98** [nextauthjs/next-auth#12106] [Help]
I'm using the GitLab OAuth provider with a database adapter and JWT session strategy.
Version: 5.0.0-beta.22

On the first sign in everything works flawlessly and the _user_ and linked gitlab _account_ are created.
However on subsequent sign ins (sign out and sign in again) next-auth throws the error  
`OAuthAccountNotLinked: Another account already exists with the same e-mail address.`.  
> Thrown when an Email address is already associated with an account but the user is trying an OAuth account that is not linked to it.

If I'm understanding the docs correctly, this error should only be thrown if the user is trying to sign in with an account **not already linked to it**. But in my case it is thrown when signin in with the same (already linked) provider account.

Am I misunderstanding something?

**Comment by k-taro56** [comment]
next-auth uses the ID provided by the OAuth provider, not the email address, to determine if the accounts are identical.
This would mean that this ID would be different the next time you log in.
